### PR TITLE
[#5230] publication dates management command

### LINKED ISSUE #5230: Management command for published resources
**Describe the feature you'd like and what it will do**
We should have a management command to report on resources published within a timeframe.
Consider also adding to our metrics system

**Additional context**
HS 2.9.2

### PR DESCRIPTION
**Pull request by devincowan**
Resolves #5230 
Requires merge of https://github.com/hydroshare/docker-elk/pull/11 and release to metrics system

### Pull Request Checklist: 
- [ ] Positive Test Case Written by Dev

- [ ] Automated Testing

- [ ] Sufficient User and Developer Documentation

- [ ] Passing Jenkins Build

- [ ] Peer Code review and approval

### Positive Test Case
1. Publish a resource
2. Run `./hsctl managepy stats --resources-details` and ensure that the publication date is included
3. Run `./hsctl managepy list_published_by_year` and see that it prints out the published resource

### CODE REVIEW

**Comment by devincowan** [comment]
@pkdash this PR is closely related to https://github.com/hydroshare/hydroshare/issues/5216
Would love to get it into [2.10](https://github.com/hydroshare/hydroshare/issues/5227) if you have a chance to review it! That way we only have to [release our metrics system once](https://github.com/deviantony/docker-elk/pull/921)

File: hs_core/management/commands/list_published_by_year.py [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+        published_date = [dt for dt in meta_dates if dt.type == "published"]
+        citation_date = None
+        if published_date:
+            citation_date = published_date[0]
+        else:
+            print(f"Publication date not found for {resource.short_id}")
+
+        return citation_date.start_date
+
+    def print_resource(self, res, pub_date):
```
Comment (pkdash) [inline hs_core/management/commands/list_published_by_year.py:75]: Should we print resource type?

**Review by pkdash — APPROVED** [review]
Left few comments. If you decide to make changes I can re-approve.

File: hs_tracking/management/commands/stats.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from django.db.models import Q
 from django.utils import timezone
 import zoneinfo
-from hs_core.models import BaseResource
+from hs_core.models import BaseResource, Date
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_tracking/management/commands/stats.py:12]: https://github.com/hydroshare/hydroshare/blob/5230-publication-dates-cmd/hs_tracking/management/commands/stats.py#L162

File: hs_core/management/commands/list_published_by_year.py [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+            dest='owned_by',
+            help='limit to resources owned by specific user'
+        )
+
+    def handle(self, *args, **options):
+        days = options['days']
+        resources = BaseResource.objects.filter(raccess__published=True)
+
+        if options['owned_by'] is not None:
+            owner = User.objects.get(username=options['owned_by'])
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/management/commands/list_published_by_year.py]: handled, thanks!

File: hs_core/management/commands/list_published_by_year.py [hunk 4]
```diff
@@ -0,0 +1,97 @@
...
+            resources.filter(r2urp__user=owner,
+                             r2urp__privilege=PrivilegeCodes.OWNER)
+
+        if options['type'] is not None:
+            resources.filter(resource_type=options['type'])
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/management/commands/list_published_by_year.py]: Good call, added that

File: hs_core/management/commands/list_published_by_year.py [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                    continue
+            if days:
+                cuttoff_time = timezone.now() - timedelta(days)
+                if not pub_date >= cuttoff_time:
+                    continue
+            self.print_resource(resource, pub_date)
+
+    def get_publication_date(self, resource):
+        meta_dates = resource.metadata.dates.all()
+        published_date = [dt for dt in meta_dates if dt.type == "published"]
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/management/commands/list_published_by_year.py]: Thanks, I made that change

File: hs_core/management/commands/list_published_by_year.py [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+        citation_date = None
```
[the remark this answers is not in the document]
Comment (devincowan) [reply hs_core/management/commands/list_published_by_year.py]: :) yeah I lifted this code block from elsewhere. Updated it, thanks

File: hs_core/management/commands/list_published_by_year.py (hunk 1 above)
Comment (devincowan) [reply hs_core/management/commands/list_published_by_year.py:79]: Why not! I added that, thanks

**Comment by devincowan** [comment]
@pkdash believe I responded to all of your comments. Thanks for the thorough look!

**Review by pkdash — APPROVED** [review]

**Pull request merged**